platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/MerkleDbStatistics.java

```diff
@@ -455,6 +458,9 @@ public void setHashesStoreCompactionTimeMs(final Integer compactionLevel, final
      */
     public void setHashesStoreCompactionSavedSpaceMb(final int compactionLevel, final double value) {
         assert compactionLevel >= 1 && compactionLevel <= config.maxCompactionLevel();
+        if (hashesStoreCompactionTimeMsList.isEmpty()) {
```

Comment:
It is not a good idea to check `hashesStoreCompactionTimeMsList` when you are going to access `hashesStoreCompactionSavedSpaceMbList`.

Suggested fix:
```
        if (compactionLevel <= hashesStoreCompactionSavedSpaceMbList.size()) {
            hashesStoreCompactionSavedSpaceMbList.get(compactionLevel - 1).update(value);
        }
```
with a possible comment that the method may be called before metrics are registered.

Reply:
I agree that the comment can be helpful but the suggested fix is not going to work because if the list is empty, it means that `register` method hasn't been called yet and, therefore `hashesStoreCompactionSavedSpaceMbList.get(compactionLevel - 1)` returns null.

Reply:
I added comments for these `if` statements.

Reply:
Agreed.

Reply:
Why is `hashesStoreCompactionTimeMsList` checked in every method? Should each method check its own metrics list? Assume in the future there are many `registerMetrics()` calls, so lists are populated independently. To me, if a method works with, e.g. `leavesStoreCompactionTimeMsList`, then it should check `leavesStoreCompactionTimeMsList`, not `hashesStoreCompactionTimeMsList`

Reply:
Yes, agreed. I fixed it.